### ISSUE TITLE
NewWithParentheses sniff incorrect detection for anonymous class with attribute

### ISSUE THREAD

**Issue by schlndh** [slevomat/coding-standard#1538]
I came across a code like this:

```php
new
#[AllowDynamicProperties]
class
{
};
```

phpcs generates this error:

```
 3 | ERROR | [x] Usage of "new" without parentheses is disallowed.
   |       |     (SlevomatCodingStandard.ControlStructures.NewWithParentheses.MissingParentheses)
``` 
If I let phpcbf fix the error, the result is this:

```php
new
#[AllowDynamicProperties]
class
{
}();
```
which is a syntax error.

It works fine without the attribute.
